### ISSUE TITLE
PetSets PV controller Error

### ISSUE THREAD

**Issue by Nalum** [kubernetes/minikube#524]
**Minikube version**: v0.8.0
**Environment**:
- **OS**:

>   PRETTY_NAME="Debian GNU/Linux 8 (jessie)"
>   NAME="Debian GNU/Linux"
>   VERSION_ID="8"
>   VERSION="8 (jessie)"
>   ID=debian
>   HOME_URL="http://www.debian.org/"
>   SUPPORT_URL="http://www.debian.org/support"
>   BUG_REPORT_URL="https://bugs.debian.org/"

- **VM Driver**: virtualbox
- **Docker version**: Docker version 1.12.1, build 23cf638

**What happened**:
Create a PetSet with `volumeClaimTemplates`, a PVC was created but a host path PV was not created.

```
$ kuDEVn describe pvc mongodb-node-0
Name:		mongodb-node-0
Namespace:	dev-nalum
Status:		Pending
Volume:		
Labels:		name=node
		role=mongodb
		version=3.2
Capacity:	
Access Modes:	
Events:
  FirstSeen	LastSeen	Count	From				SubobjectPath	Type		Reason			Message
  ---------	--------	-----	----				-------------	--------	------			-------
  5m		12s		25	{persistentvolume-controller }			Warning		ProvisioningFailed	No provisioner plugin found for the claim!
```

**What you expected to happen**:
Expected a host path PV to be created to fill the PVC based on this: https://github.com/kubernetes/kubernetes/blob/v1.3.5/examples/experimental/persistent-volume-provisioning/README.md

Specifically:
>  A fourth provisioner is included for testing and development only. It creates HostPath volumes, which will never work outside of a single node cluster. It is not supported in any way except for local for testing and development.

Perhaps this only means for testing and development of kubernetes?

To enable it requires setting the kubernetes flag `--enable-hostpath-provisioner=true`

**How to reproduce it**:
Create the following PetSet:

```yaml
---

apiVersion: apps/v1alpha1
kind: PetSet
metadata:
  name: node
  labels:
    name: node
    version: "3.2"
    role: mongodb
spec:
  replicas: 1
  serviceName: mongodb
  template:
    metadata:
      name: node
      labels:
        version: "3.2"
        name: node
        role: mongodb
      annotations:
        prometheus.io/scrape: "true"
        prometheus.io/port: "9001"
        pod.alpha.kubernetes.io/initialized: "true"
    spec:
      terminationGracePeriodSeconds: 0
      containers:
      - name: node
        image: mongo:3.2
        env:
        - name: NAMESPACE
          valueFrom:
            fieldRef:
              fieldPath: metadata.namespace
        command:
        - mongod
        args:
        - --replSet
        - dev
        - --bind_ip
        - 0.0.0.0
        ports:
        - containerPort: 27017
          name: node
        volumeMounts:
        - name: mongodb
          mountPath: /data
  volumeClaimTemplates:
  - metadata:
      name: mongodb
      annotations:
        volume.alpha.kubernetes.io/storage-class: anything
    spec:
      accessModes: [ "ReadWriteOnce" ]
      resources:
        requests:
          storage: 6Gi
```

**Comment by dlorenc** [comment]
Seems like we can enable this flag in the controller-manager with something like this:

`config.VolumeConfiguration.EnableDynamicProvisioning = true`

Does anyone know if there are downsides to just enabling this always?

**Comment by Nalum** [comment]
It would also be good to make sure that the directory the PV creates are persisted.

I'm not sure if there is a way to specify a path for the Dynamic Provisioning to use.

**Comment by dlorenc** [comment]
@Nalum good idea. Do you know where it typically creates the PVs?

**Comment by Nalum** [comment]
I do not. I've only started looking at petsets. If I get a bit of time I'll find out over the weekend.

**Comment by r2d4** [comment]
Needs this commit to be cherry picked https://github.com/kubernetes/kubernetes/pull/32582

**Comment by dlorenc** [comment]
Or we can change the config we pass to localkube now to include all the strings we need.

**Comment by dlorenc** [comment]
This should now be available on 1.4 which is defailt in 0.11
